language_service/src/completion.rs

```diff
@@ -121,6 +131,16 @@ fn expected_word_kinds(
     }
 }
 
+fn in_comment(source_contents: &str, cursor_offset: u32) -> bool {
+    // find the last newline before the cursor
+    let last_line_start = source_contents[..cursor_offset as usize]
+        .rfind('\n')
+        .unwrap_or(0);
+    // find the last comment start before the cursor
+    let last_comment_start = source_contents[last_line_start..cursor_offset as usize].rfind("//");
```

Comment:
Is it possible this logic could get confused by string literals, as in:
```qsharp
let x = "// this is in a string" + StrUtils.|
```

Reply:
Well, that line doesn't start with `//` (it might've been a GitHub antic). I think you meant something like,

```qsharp
let x = "foo
// this is in a string" + StrUtils.|
```

Reply:
Per side discussion - this is a valid concern, but requires a bit larger fix. Instead of patching up this case, I'd rather spend a bit more time to look at how we fix the root cause (in the lexer itself). I'll open an issue for it.